libbeat/publisher/queue/memqueue/eventloop.go

```diff
@@ -143,16 +145,18 @@ func (l *directEventLoop) handleMetricsRequest(req *metricsRequest) {
 func (l *directEventLoop) insert(req *pushRequest) {
 	log := l.broker.logger
 
-	st := req.state
-	if st == nil {
-		l.buf.insert(req.event, clientState{})
-	} else if st.cancelled {
+	if req.producer != nil && req.producer.state.cancelled {
 		reportCancelledState(log, req)
 	} else {
-		l.buf.insert(req.event, clientState{
-			seq:   req.seq,
-			state: st,
-		})
+		if req.resp != nil {
```

Comment:
I think the comment to this `insert` function is outdated since it does not actually return anything and I think it needs a comment inside the function to explain why `req.resp` can be `nil` and how we handle it.

Reply:
`req.resp` actually can't be nil! That check is leftover from an earlier iteration where it could be, so I removed it.